specification/keyvault/Keyvault.Management/models.tsp

```diff
@@ -0,0 +1,2488 @@
+   */
+  id: string;
+}
+
+/**
+ * A region that this managed HSM Pool has been extended to.
+ */
+model MhsmGeoReplicatedRegion {
+  /**
+   * Name of the geo replicated region.
+   */
+  name?: string;
+
+  /**
+   * Provisioning state of the geo replicated region.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: GeoReplicationRegionProvisioningState;
+
+  /**
+   * A boolean value that indicates whether the region is the primary region or a secondary region.
+   */
+  isPrimary?: boolean;
+}
+
+/**
+ * Private endpoint connection item.
+ */
+model MhsmPrivateEndpointConnectionItem {
+  /**
+   * Id of private endpoint connection.
+   */
+  id?: string;
+
+  /**
+   * Modified whenever there is a change in the state of private endpoint connection.
+   */
+  etag?: string;
+
+  /**
+   * Private endpoint connection properties.
+   */
+  #suppress "@azure-tools/typespec-azure-core/no-private-usage" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+  properties?: MhsmPrivateEndpointConnectionProperties;
+}
+
+/**
+ * Properties of the private endpoint connection resource.
+ */
+model MhsmPrivateEndpointConnectionProperties {
+  /**
+   * Properties of the private endpoint object.
+   */
+  privateEndpoint?: MhsmPrivateEndpoint;
+
+  /**
+   * Approval state of the private link connection.
+   */
+  privateLinkServiceConnectionState?: MhsmPrivateLinkServiceConnectionState;
+
+  /**
+   * Provisioning state of the private endpoint connection.
+   */
+  @visibility(Lifecycle.Read)
+  provisioningState?: PrivateEndpointConnectionProvisioningState;
+}
+
+/**
+ * Private endpoint object properties.
+ */
+model MhsmPrivateEndpoint {
+  /**
+   * Full identifier of the private endpoint resource.
+   */
+  @visibility(Lifecycle.Read)
+  id?: string;
+}
+
+/**
+ * An object that represents the approval state of the private link connection.
+ */
+model MhsmPrivateLinkServiceConnectionState {
+  /**
+   * Indicates whether the connection has been approved, rejected or removed by the key vault owner.
+   */
+  status?: PrivateEndpointServiceConnectionStatus;
+
+  /**
+   * The reason for approval or rejection.
+   */
+  description?: string;
+
+  /**
+   * A message indicating if changes on the service provider require any updates on the consumer.
+   */
+  actionsRequired?: ActionsRequired;
+}
+
+/**
+ * The security domain properties of the managed hsm.
+ */
+#suppress "@azure-tools/typespec-azure-core/casing-style" "FIXME: Update justification, follow aka.ms/tsp/conversion-fix for details"
+model ManagedHSMSecurityDomainProperties {
+  /**
+   * Activation Status
+   */
+  @visibility(Lifecycle.Read)
+  activationStatus?: ActivationStatus;
+
+  /**
+   * Activation Status Message.
+   */
+  @visibility(Lifecycle.Read)
+  activationStatusMessage?: string;
+}
+
+/**
+ * Managed HSM resource
+ */
+model ManagedHsmResource {
```

Comment:
The current service both exist self-defined systemData and common-types systemData. Here are the examples:
`DeletedManagedHsm` is common type one: typspec link https://github.com/Azure/azure-rest-api-specs/blob/ba8007ddc9335e47a90aec28efa39ed976c46590/specification/keyvault/Keyvault.Management/main.tsp#L40


`ManagedHsmResource` is self-defined one: typespec link https://github.com/Azure/azure-rest-api-specs/blob/ba8007ddc9335e47a90aec28efa39ed976c46590/specification/keyvault/Keyvault.Management/models.tsp#L1835
original swagger link https://github.com/Azure/azure-rest-api-specs/blob/main/specification/keyvault/resource-manager/Microsoft.KeyVault/stable/2024-11-01/managedHsm.json#L1117